### PR TITLE
Refactor sol-cov to avoid keeping traceInfo in memory

### PR DESCRIPTION
**Pull request by albrow**
## Description

We were running into problems with sol-cov where it couldn't run coverage over all the contracts tests at once because Node was running out of memory. This PR fixes that problem by computing coverage on the fly for each `traceInfo`. This means we don't have to keep `traceInfo` in memory.

## Types of changes

*   [x] Bug fix (non-breaking change which fixes an issue)
*   [ ] New feature (non-breaking change which adds functionality)
*   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

*   [ ] Change requires a change to the documentation.
*   [ ] Added tests to cover my changes.
*   [ ] Added new entries to the relevant CHANGELOG.jsons.
*   [ ] Labeled this PR with the 'WIP' label if it is a work in progress.
*   [ ] Labeled this PR with the labels corresponding to the changed package.

### CODE REVIEW

File: packages/sol-cov/src/coverage_manager.ts [hunk 1]
```diff
@@ -130,49 +131,39 @@ export class CoverageManager {
...
-                continue;
-            }
-            const bytecodeHex = stripHexPrefix(bytecode);
-            const sourceMap = isContractCreation ? contractData.sourceMap : contractData.sourceMapRuntime;
-            const pcToSourceRange = parseSourceMap(
-                contractData.sourceCodes,
-                sourceMap,
-                bytecodeHex,
-                contractData.sources,
+    public async computeCoverageAsync(traceInfo: TraceInfo): Promise<void> {
```
Comment (LogvinovLeon) [inline packages/sol-cov/src/coverage_manager.ts]: computeSingleTraceCoverageAsync

File: packages/sol-cov/src/coverage_subprovider.ts [hunk 2]
```diff
@@ -25,12 +26,13 @@ export class CoverageSubprovider extends TraceCollectionSubprovider {
         super(defaultFromAddress, traceCollectionSubproviderConfig);
         this._coverageManager = new CoverageManager(artifactAdapter, isVerbose);
     }
+    public async handleTraceInfoAsync(traceInfo: TraceInfo): Promise<void> {
+        return this._coverageManager.computeCoverageAsync(traceInfo);
```
Comment (LogvinovLeon) [inline packages/sol-cov/src/coverage_subprovider.ts]: await  instead of return

File: packages/sol-cov/src/coverage_subprovider.ts [hunk 3]
```diff
@@ -25,12 +26,13 @@ export class CoverageSubprovider extends TraceCollectionSubprovider {
...
+    }
     /**
      * Write the test coverage results to a file in Istanbul format.
      */
     public async writeCoverageAsync(): Promise<void> {
-        const traceInfos = this.getCollectedTraceInfos();
-        _.forEach(traceInfos, traceInfo => this._coverageManager.appendTraceInfo(traceInfo));
-        await this._coverageManager.writeCoverageAsync();
+        return this._coverageManager.writeCoverageAsync();
```
Comment (LogvinovLeon) [inline packages/sol-cov/src/coverage_subprovider.ts]: await instead of return

File: packages/sol-cov/src/profiler_subprovider.ts [hunk 4]
```diff
@@ -25,12 +26,13 @@ export class ProfilerSubprovider extends TraceCollectionSubprovider {
         super(defaultFromAddress, traceCollectionSubproviderConfig);
         this._profilerManager = new ProfilerManager(artifactAdapter, isVerbose);
     }
+    public async handleTraceInfoAsync(traceInfo: TraceInfo): Promise<void> {
+        return this._profilerManager.computeCoverageAsync(traceInfo);
```
Comment (LogvinovLeon) [inline packages/sol-cov/src/profiler_subprovider.ts]: await instead of return

File: packages/sol-cov/src/profiler_subprovider.ts [hunk 5]
```diff
@@ -25,12 +26,13 @@ export class ProfilerSubprovider extends TraceCollectionSubprovider {
...
+    }
     /**
      * Write the test profiler results to a file in Istanbul format.
      */
     public async writeProfilerOutputAsync(): Promise<void> {
-        const traceInfos = this.getCollectedTraceInfos();
-        _.forEach(traceInfos, traceInfo => this._profilerManager.appendTraceInfo(traceInfo));
-        await this._profilerManager.writeProfilerOutputAsync();
+        return this._profilerManager.writeProfilerOutputAsync();
```
Comment (LogvinovLeon) [inline packages/sol-cov/src/profiler_subprovider.ts]: await instead of return

**Comment by albrow** [comment]
@LogvinovLeon I believe I've addressed all your feedback.

**Pull request merged**